From foo@baz Mon Jul 27 09:17:51 PDT 2009
Date: Mon, 27 Jul 2009 09:17:51 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: hv: force hyper-v drivers to be built as a module

From: Greg Kroah-Hartman <gregkh@suse.de>

Right now they can not be built into the kernel due to global
symbol name conflicts that this code is causing.

Thanks to Nicolas Palix <npalix@diku.dk> for pointing this out.

Reported-by: Nicolas Palix <npalix@diku.dk>
Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/hv/Kconfig |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/staging/hv/Kconfig
+++ b/drivers/staging/hv/Kconfig
@@ -1,6 +1,6 @@
 config HYPERV
 	tristate "Microsoft Hyper-V client drivers"
-	depends on X86
+	depends on X86 && m
 	default n
 	help
 	  Select this option to run Linux as a Hyper-V client operating
@@ -8,21 +8,21 @@ config HYPERV
 
 config HYPERV_STORAGE
 	tristate "Microsoft Hyper-V virtual storage driver"
-	depends on SCSI
+	depends on SCSI && m
 	default n
 	help
 	 Select this option to enable the Hyper-V virtual storage driver.
 
 config HYPERV_BLOCK
 	tristate "Microsoft Hyper-V virtual block driver"
-	depends on BLOCK
+	depends on BLOCK && m
 	default n
 	help
 	  Select this option to enable the Hyper-V virtual block driver.
 
 config HYPERV_NET
 	tristate "Microsoft Hyper-V virtual network driver"
-	depends on NET
+	depends on NET && m
 	default n
 	help
 	  Select this option to enable the Hyper-V virtual network driver.
